sdk/core/Azure.Core/src/Shared/GuidUtilities.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Buffers.Binary;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Azure.Core.Shared;
+
+/// <summary>
+/// The purpose of this class is to provide an allocation free way to parse and write Guids on .NET Standard 2.0, taking into account
+/// endianness.
+/// </summary>
+internal static class GuidUtilities
+{
+    private const int GuidSizeInBytes = 16;
+
+    public static Guid ParseGuidBytes(ReadOnlyMemory<byte> bytes)
```

Comment:
This should take ROS, not ROM, so that I can parse a guid from a stack allocated buffer, for example.

Reply:
I was actually thinking the same when I introduced my implementation changes but left it as is since the method signature was suggested like that based on the service bus usage. Now that it is in the core I think there are even stronger reasons to make this ROS instead of ROM